data.tf

```diff
@@ -9,6 +9,22 @@ data "aws_kms_alias" "ssm" {
   name = var.ssm_kms_alias
 }
 
+data "aws_ssm_parameter" "ssm_secret" {
+  for_each = var.secret_environment_variables
+  name     = "${var.parameter_path_prefix}/${each.value}"
+}
+
+## Mapping : SSM path => version ( to detect the change in particular path's value)
```

Comment:
Create map of "parameter_name => parameter_version" to detect the change  of parameter's value